### PR TITLE
Decrease running time in filtering users

### PR DESCRIPTION
**Pull request by regiskuckaertz**
Checking whether a user account is in the developer tier requires checking the keys table. Right now this is done via one query for each user, that is terrible for performance (in fact, the cleanup lambda times out on a dry run).

This PR improves the running time by doing one query per 100 users at a time. This can be achieved due [an upcoming change in Scanamo](https://github.com/scanamo/scanamo/pull/225) to add support for the `in` operator in DynamoDB. I'm hoping I can it merged soon so I can move forward with this one.

The first two commits can be ignored, it is just formatting.

(Thanks @hosamaly for spotting the issue)

### CODE REVIEW

**Comment by tomrf1** [comment]
👍 sounds sensible

File: src/main/scala/com.gu.gibbons/services/BonoboService.scala [hunk 1]
```diff
@@ -17,7 +17,8 @@ trait BonoboService[F[_]] {
    */
   def getUsers(jadis: Instant): F[Vector[User]]
 
-  def isDeveloper(user: User): F[Boolean]
+  /** Filters a list of users to those who are developers only */
+  def isDeveloper(users: Vector[User]): F[Vector[User]]
```
Comment (hosamaly) [inline src/main/scala/com.gu.gibbons/services/BonoboService.scala]: I think that we should change the name of this method. Methods starting with `is` are, conventionally, predicates.

How about `getDevelopers` or `filterDevelopers`?

File: src/main/scala/com.gu.gibbons/services/interpreters/BonoboInterpreter.scala [hunk 2]
```diff
@@ -30,21 +32,28 @@ class BonoboInterpreter(config: Settings,
     for {
       _ <- logger.info(s"Getting all the users created $jadis ago")
       millis = jadis.toEpochMilli
-      users <- getUsersMatching(
-        not(attributeExists('remindedAt)) and ('extendedAt <= millis or (not(attributeExists('extendedAt)) and 'createdAt <= millis))
-      )
+      users <- getItems(usersTable,
+                        not(attributeExists('remindedAt)) and ('extendedAt <= millis or (not(
+                          attributeExists('extendedAt)
+                        ) and 'createdAt <= millis)))
```
Comment (hosamaly) [inline src/main/scala/com.gu.gibbons/services/interpreters/BonoboInterpreter.scala]: This condition is hard to read. Would you mind explaining it in plain English?

Comment (regiskuckaertz) [reply src/main/scala/com.gu.gibbons/services/interpreters/BonoboInterpreter.scala]: It is a translation of the logical assertion _¬p ∧ (q ∧ r ∨ ¬q ∧ s)_ where `q ∧ r` can be simplified to just `r` because `r ⇒ q`.

**Comment by regiskuckaertz** [comment]
Good to know: in the AWS documentation, it is mentioned that the IN operator will accept up to 100 values to match against. My tests have demonstrated that the correct number if _100 - n_ where _n_ is the number of sub-expressions in your filter. In my case, I had
```
'tier -> "Developer" and 'bonoboId -> _set of 100 values_
```
and it failed with a message saying I had 101 operands.
```
'tier -> "Developer" and 'bonoboId -> _set of 99 values_
```
worked perfectly. I have tested in CODE using PROD resources, it's not timing out anymore and only developer accounts are targeted.
